main/githooks.py

```diff
@@ -104,22 +147,46 @@ def get_commit_files():
     return result
 
 
+def parse_diff_header(header_line):
+    changed_lines = []
+    match = parse_diff_header.pattern.match(header_line)
+    start = int(match.group(1))
+    if match.group(2):
+        for num in range(int(match.group(3))):
+            changed_lines.append(start + num)
+    else:
+        changed_lines.append(start)
+    return changed_lines
```

Comment:
I slightly worry what will happen here if the file is huge and this array ends up containing a million line numbers or more.  It might be worth storing a list of ranges rather than lists of individual line numbers.

Reply:
It's good to see this has some tests

Reply:
Good point.

Reply:
Please check https://github.com/ccdc-opensource/commit-hooks/pull/3